airflow/providers/amazon/aws/executors/ecs/Setup_guide.md

```diff
@@ -0,0 +1,148 @@
+
+**NOTE**: Be sure that the status of your DB is Available before testing connectivity
+
+```
+psql -h <endpoint> -p 5432 -U <username> <db_name>
+```
+
+The endpoint can be found on the "Connectivity and Security" tab, the username (and password) are the credentials used when creating the database.
+The db_name should be `airflow_db` (unless a different one was used when creating the database.)
+
+You will be prompted to enter the password if the connection is successful.
+
+
+## Creating an ECS Cluster with Fargate, and Task Definitions
+
+In order to create a Task Definition for the ECS Cluster that will work with Apache Airflow, you will need a Docker image that is properly configured. See the [Dockerfile](README.md#dockerfile-for-ecs-executor) section for instructions on how to do that.
+
+Once the image is built, it needs to be put in a repository where it can be pulled by ECS. There are multiple ways to accomplish this. This guide will go over doing this using Amazon Elastic Container Registry (ECR).
+
+### Create an ECR Repository
+
+1. Log in to your AWS Management Console and navigate to the ECR service.
+2. Click Create repository.
+3. Name the repository and fill out other information as required.
+4. Click Create Repository.
+5. Once the repository has been created, click on the repository. Click on the "View push commands" button on the top right.
+6. Follow the instructions to push the Docker image, replacing image names as appropriate. Ensure the image is uploaded by refreshing the page once the image is pushed.
+
+### Create ECS Cluster
+
+1. Log in to your AWS Management Console and navigate to the Amazon Elastic Container Service.
+2. Click "Clusters" then click "Create Cluster".
+3. Make sure that AWS Fargate (Serverless) is selected under Infrastructure.
+4. Select other options as required and click Create to create the cluster.
+
+### Create Task Definition
+
+1. Click on Task Definitions on the left hand bar, and click Create new task definition.
+2. Choose the Task Definition Family name. Select AWS Fargate for the Launch Type.
+3. Select or create the Task Role and Task Execution Role, and ensure the roles have the required permissions to accomplish their respective tasks. You can choose to create a new Task Execution role that will have the basic minimum permissions in order for the task to run.
+4. Select a name for the container, and use the image URI of the image that was pushed in the previous section. Make sure the role being used has the required permissions to pull the image.
+5. Add the following environment variables to the container:
+
+ - `AIRFLOW__DATABASE__SQL_ALCHEMY_CONN`, with the value being the PostgreSQL connection string in the following format using the values set during the [Database section](#create-the-rds-db-instance) above:
+
```

Comment:
Maybe we can teach users how to do it well (how do it bad they found themselves 🤣 ) and recommend to use for secrets builtin feature of ECS: grab secret environments from SSM or SM?
- https://docs.aws.amazon.com/AmazonECS/latest/userguide/secrets-envvar-ssm-paramstore.html#secrets-envvar-ssm-paramstore-update-container-definition
- https://docs.aws.amazon.com/AmazonECS/latest/userguide/secrets-envvar-secrets-manager.html#secrets-examples

Reply:
I think once we have some more robust documentation in a followup PR. This would be good to add. For now, we're trying to keep it as simple as possible and assuming any folks using this for production are aware of common practices for managing secrets in their production infra.

Reply:
_Expectation_

>assuming any folks using this for production are aware of common practices for managing secrets in their production infra.

_Reality_ 

**Ctrl (⌘) + C** -> **Ctrl (⌘) + V** -> "Now it is our production ready solution" 🤣